src/node/mod.rs

```diff
@@ -653,10 +653,19 @@ impl Node {
                 _ => unreachable!(),
             },
             Stage::Approved(stage) => match msg.variant() {
-                Variant::NeighbourInfo { elders_info, .. } => {
+                Variant::NeighbourInfo {
+                    elders_info,
+                    section_key,
+                    nonce,
+                } => {
                     msg.dst().check_is_section()?;
-                    let src_key = *msg.proof_chain_last_key()?;
-                    stage.handle_neighbour_info(&mut self.core, elders_info.clone(), src_key)?;
+                    stage.handle_neighbour_info(
+                        &mut self.core,
+                        elders_info.value.clone(),
+                        *section_key,
+                        *msg.dst_key(),
+                        *nonce,
```

Comment:
The `nonce` should be derived from the message that triggered sending the `NeighbourInfo`. in this case that's the incoming `NeighbourInfo` message itself, not the message that triggered it. So this should be `msg.hash()`.

Reply:
I thought as this is a B->A NeighbourInfo triggered by a A-B NeighbourInfo, and the elders_info is different, so using the ame nonce can `safely` link them.
I am ok with your suggestion, will change it.